tests/scripts/test_generate_pipeline.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+
+
+import pathlib
+import tempfile
+import typing
+
+import pytest
+from click.testing import CliRunner
+from ruamel import yaml
+
+from scripts import generate_dag, generate_pipeline_files, generate_terraform
+from scripts.generate_pipeline_files import create_pipeline
+
+yaml = yaml.YAML(typ="safe")
+
+PROJECT_ROOT = generate_pipeline_files.PROJECT_ROOT
+DATASETS_PATH = PROJECT_ROOT / "datasets"
+
+SAMPLE_YAML_PATHS = {
+    "dataset": PROJECT_ROOT / "samples" / "dataset.yaml",
+    "pipeline": PROJECT_ROOT / "samples" / "pipeline.yaml",
+}
+
+ENV_PATH = generate_pipeline_files.PROJECT_ROOT / ".test"
+ENV_DATASETS_PATH = ENV_PATH / "datasets"
+
+
+@pytest.fixture
+def env() -> str:
+    return "test"
+
+
+@pytest.fixture
+def dataset_path() -> typing.Iterator[pathlib.Path]:
+    with tempfile.TemporaryDirectory(
+        dir=generate_dag.DATASETS_PATH, suffix="_dataset"
+    ) as dir_path:
+        yield pathlib.Path(dir_path)
+
+
+@pytest.fixture
+def pipeline_path(
+    dataset_path: pathlib.Path, suffix="_pipeline"
+) -> typing.Iterator[pathlib.Path]:
+    pipelines_dir = dataset_path / "pipelines"
+    pipelines_dir.mkdir(parents=True, exist_ok=True)
+    with tempfile.TemporaryDirectory(dir=pipelines_dir, suffix=suffix) as dir_path:
+        yield pathlib.Path(dir_path)
+
+
+def all_pipelines() -> typing.Iterator[typing.Tuple[pathlib.Path, pathlib.Path]]:
+    for dataset_path_ in generate_terraform.list_subdirs(generate_dag.DATASETS_PATH):
+        for pipeline_path_ in generate_terraform.list_subdirs(
+            dataset_path_ / "pipelines"
+        ):
+            yield dataset_path_, pipeline_path_
+
+
+test_flow = {
+    "friendly_dataset_name": "my friendly dataset_description",
+    "resource_needed1": "y",
+    "resource1": "bq",
+    "bq_description": "dataset.yaml bq description",
+    "resource_needed2": "y",
+    "resource2": "gcs",
+    "gcs_bucket_name": "my-pipeline-test-bucket",
+    "gcs_bucket_location": "US",
+    "resource_needed3": "n",
+    "bq_tables": "table1, table2, table3",
+    "operators": "BashOperator",
+    "add_another_operator": "n",
+}
+
+
+def test_pipeline_directory():
```

Comment:
Use function names that better describe what's being tested. Here's an example https://github.com/GoogleCloudPlatform/public-datasets-pipelines/blob/0c285639b9f8c7fae10581286bb0df86a47eee8c/tests/scripts/test_generate_dag.py#L169

Do this as well for the rest of the tests